tiled/_tests/test_authentication.py

```diff
@@ -88,6 +90,40 @@ def test_password_auth(enter_password, config):
                 from_context(context, username="alice")
 
 
+def test_logout(enter_password, config, tmpdir):
+    """
+    Logging out revokes the session, such that it cannot be refreshed.
+    """
+    with Context.from_app(build_app_from_config(config)) as context:
+        # Log in as Alice.
+        with enter_password("secret1"):
+            from_context(context, username="alice")
+        # Reuse token from cache.
+        client = from_context(context, username="alice")
+        # This was set to a unique (temporary) dir by an autouse fixture in conftest.py.
+        tiled_cache_dir = os.environ["TILED_CACHE_DIR"]
+        # Make a backup copy of the cache directory, which contains the auth tokens.
+        shutil.copytree(tiled_cache_dir, tmpdir / "backup")
+        # Logout does two things:
+        # 1. Revoke the session, so that it cannot be refreshed.
+        # 2. Clear the tokens related to this session from in-memory state
+        #    and on-disk state.
+        client.logout()
+        # Restore the tokens from backup.
+        # Our aim is to test, below, that even if you have the tokens they
+        # can't be used anymore.
+        shutil.rmtree(tiled_cache_dir)
+        shutil.copytree(tmpdir / "backup", tiled_cache_dir)
+        # There is no way to revoke a JWT access token. It expires after a
+        # short time window (minutes) but it will still work here, as it has
+        # not been that long.
+        client = from_context(context, username="alice")
+        # The refresh token refers to a revoked session, so refreshing the
+        # session to new a *new* access and refresh token will fail.
+        with pytest.raises(CannotRefreshAuthentication):
+            client.context.force_auth_refresh()
```

Comment:
This test was written first, and before the changes, this line was able to refresh the session _without_ raising an error. This reproduced #602 and demonstrated that the PR fixed it.